classes/timeslot_manager.php

```diff
@@ -364,9 +368,9 @@ public static function timeslot_signup(int $observationid, int $slotid, int $use
         $signedupslot = self::get_registered_timeslot($observationid, $userid);
 
         if ($timeslot->observee_id !== null) {
-            throw new moodle_exception("Could not signup to timeslot. Timeslot already taken.");
+            throw new \moodle_exception("Could not signup to timeslot. Timeslot already taken.");
```

Comment:
I think instead of exceptions here, we can be a little nicer to the user, and instead just use \core\notification::error and reload the page. also move this to a lang string